sdk/core/azure-core/src/exception.cpp

```diff
@@ -21,25 +21,36 @@ namespace Azure { namespace Core {
 
   RequestFailedException::RequestFailedException(
       std::unique_ptr<Azure::Core::Http::RawResponse>& rawResponse)
-      : RequestFailedException("Received an HTTP unsuccessful status code.")
+      : std::runtime_error(GetRawResponseErrorMessage(rawResponse)),
+        RawResponse(std::move(rawResponse)),
+        // These are guaranteed to always be present in the rawResponse.
+        StatusCode(RawResponse->GetStatusCode()), ReasonPhrase(RawResponse->GetReasonPhrase()),
+        // The response body may or may not have these fields
+        ErrorCode(GetRawResponseField(RawResponse, "code")),
+        Message(GetRawResponseField(RawResponse, "message"))
```

Comment:
> Also, consider adding a changelog entry to mention that the FailedRequestException what() is now enhanced to provide more information from the http response

That makes a TON of sense I'll make that happen.

Reply:
We can, but the instant we hit the code which initializes the `RawResponse` member, the `rawResponse` member becomes inert. I moved that earlier to ensure that we had a stable value we could use throughout the member initializers.

`RawResponse` needs to move regardless - it can be first, or it can be last. It doesn't really matter to me which it was, I thought using `RawResponse` throughout the ctor made sense since that's how we expect that field to be used throughout the rest of the class.